docs(cli): update CLI documentation in README (#314)
code-pushup/cli: packages/cli/README.md

      - run: npx code-pushup autorun --upload.apiKey=${{ secrets.PORTAL_API_KEY }}
```

## CLI commands and options

### Global Options

| Option           | Type      | Default                 | Description                                                            |
| ---------------- | --------- | ----------------------- | ---------------------------------------------------------------------- |
| **`--progress`** | `boolean` | `true`                  | Show progress bar in stdout.                                           |
| **`--verbose`**  | `boolean` | `false`                 | When true creates more verbose output. This is helpful when debugging. |
| **`--config`**   | `string`  | `code-pushup.config.js` | Path to the config file, e.g. code-pushup.config.js                    |

### Common Command Options

| Option                      | Type                 | Default  | Description                                          |
| --------------------------- | -------------------- | -------- | ---------------------------------------------------- |
| **`--persist.outputDir`**   | `string`             | n/a      | Directory for the produced reports.                  |

> | **`--persist.filename`**    | `string`             | `report` | Filename for the produced reports without extension. |

Reviewer: I noticed you are using reports in plural in several places. As the ClI only produces one report at a time, I think a singular "report" should be used.

Reviewer 2: There is only one report, but can be multiple report files (`report.json` and `report.md`).

| **`--persist.format`**      | `('json' \| 'md')[]` | `json`   | Format(s) of the report file.                        |
| **`--upload.organization`** | `string`             | n/a      | Organization slug from portal.                       |
| **`--upload.project`**      | `string`             | n/a      | Project slug from portal.                            |
| **`--upload.server`**       | `string`             | n/a      | URL to your portal server.                           |
| **`--upload.apiKey`**       | `string`             | n/a      | API key for the portal server.                       |

> [!NOTE]  

> > All common options can be specified in the configuration file as well. CLI arguments take precedence over configuration file options.

Reviewer 2: Perfect :ok_hand: Also, I don't know GitHub's Markdown supports this `[!NOTE]` syntax, very nice :+1:

### Commands

#### `collect` command

> Run commands over:

Reviewer 2: I don't think this grammar works :confused: How about just "Usage:"?

`code-pushup collect [options]`

Description:  

> Run plugins and collect results.

Reviewer: If I read the description without the previous knowledge of the project, I would not understand which command to use to produce a report - I suggest adding that this command produces a report based on the collected results as well.

      - run: npm ci
      - run: npx code-pushup autorun --upload.apiKey=${{ secrets.PORTAL_API_KEY }}
```

> # CLI

Reviewer 2: There is already a heading level 1 (`#`) at the top of this document, so this should be level 2 (`##`), and the sub-sections should then be nested accordingly (`##` :arrow_right: `###`, etc.).

(We could integrate [markdownlint](https://github.com/DavidAnson/markdownlint) to these catch typograhic errors, BTW. I personally use the VSCode extension.)

Also, maybe "CLI reference" or "CLI commands and options" would help distinguish this section from what's above? :thinking:

Author: Agree. 

Create issue #320 about markdownlint

## Global Options

| Option           | Type      | Default                 | Description                                                            |
| ---------------- | --------- | ----------------------- | ---------------------------------------------------------------------- |
| **`--progress`** | `boolean` | `true`                  | Show progress bar in stdout.                                           |
| **`--verbose`**  | `boolean` | `false`                 | When true creates more verbose output. This is helpful when debugging. |

> | **`--config`**   | `string`  | `code-pushup.config.js` | Path the the config file, e.g. code-pushup.config.js                   |

Reviewer 3: ```suggestion
| **`--config`**   | `string`  | `code-pushup.config.js` | Path to the config file, e.g. code-pushup.config.js                   |
```

## Commands

### `collect` command

Run commands over:
`code-pushup collect [options]`

Description:  

> Run Plugins and collect results.

Reviewer 2: ```suggestion
Run plugins and collect results.
```

| Option                    | Type       | Default | Description                                     |
| ------------------------- | ---------- | ------- | ----------------------------------------------- |
| **`--onlyPlugins`**       | `string[]` | `[]`    | Only run the specified plugins.                 |

> | **`--persist.outputDir`** | `string`   | n/a     | Directory for the produced reports.             |

Reviewer 2: Maybe out of scope of this PR, but why don't we specify a default location for `outputDir`? E.g. `'.code-pushup'`.

Reviewer 3: Also thought about it!

Author: @matejchalk created #321

> | **`--persist.filename`**  | `string`   | n/a     | Filename for the produced reports.              |

Reviewer 2: Maybe good to specify "without extension"?

> | **`--persist.filename`**  | `string`   | n/a     | Filename for the produced reports.              |

Reviewer 2: ```suggestion
| **`--persist.filename`**  | `string`   | `'report'`    | Filename for the produced reports.              |
```

> | **`--persist.format`**    | `enum`     | n/a     | Format of the report output. e.g. `md`, `json`. |

Reviewer 2: ```suggestion
| **`--persist.format`**    | `('json' | 'md')[]`     | 'json'     | Format(s) of the report file. |
```

### `upload` command

Run commands over:
`code-pushup upload [options]`

Description:
Upload reports to the Code PushUp portal.

| Option                      | Type     | Default | Description                                     |
| --------------------------- | -------- | ------- | ----------------------------------------------- |
| **`--persist.outputDir`**   | `string` | n/a     | Directory for the produced reports.             |
| **`--persist.filename`**    | `string` | n/a     | Filename for the produced reports.              |
| **`--persist.format`**      | `enum`   | n/a     | Format of the report output. e.g. `md`, `json`. |
| **`--upload.organization`** | `string` | n/a     | Organization slug from portal.                  |
| **`--upload.project`**      | `string` | n/a     | Project slug from portal.                       |
| **`--upload.server`**       | `string` | n/a     | URL to your portal server.                      |

> | **`--upload.apiKey`**       | `string` | n/a     | API key for the portal server.                  |

Reviewer 2: Do we need to specify these 4 times? They're always the same, so it feels like a waste of space.

Instead, I would explain that these options are read from the config file, so the CLI arguments act as an override.

### `autorun` command

Run commands over:
`code-pushup autorun [options]`

Description:

> Run Plugins, collect results and upload reports to the Code PushUp portal.

Reviewer 2: ```suggestion
Run plugins, collect results and upload report to the Code PushUp portal.
```
